### PR TITLE
Added support for in-memory realms

### PR DESCRIPTION
**Pull request by astigsen**
## What, How & Why?

This adds support for in-memory realms.

Reference to the issue(s) addressed by this PR: N/A

## ☑️ ToDos

* [x]  Changelog entry
* [x]  Tests
* [x]  Public documentation PR created or is not necessary (see https://github.com/realm/realm-js/issues/1308)
* [ ]  `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: docs/realm.js [hunk 1]
```diff
@@ -277,6 +277,11 @@ Realm.defaultPath;
  *   will be skipped if another process is accessing it.
  * @property {string} [path={@link Realm.defaultPath}] - The path to the file where the
  *   Realm database should be stored.
+ * @property {boolean} [inMemory=false] - Specifies if this Realm should be opened in-memory. This
+ *    still requires a path (can be the default path) to identify the realm so other processes can
```
Comment (kneth) [inline docs/realm.js]: "the realm" -> "the Realm"

File: docs/realm.js [hunk 2]
```diff
@@ -277,6 +277,11 @@ Realm.defaultPath;
...
+ *    open the same realm. The file will also be used as swap space if the realm becomes bigger than
```
Comment (kneth) [inline docs/realm.js]: "same realm" -> "same Realm"
"the realm" -> "the Realm"

File: docs/realm.js [hunk 3]
```diff
@@ -277,6 +277,11 @@ Realm.defaultPath;
...
+ *    what fits in memory, but it is not persistent and will be removed when the last instance
```
Comment (kneth) [inline docs/realm.js]: "persistent" -> "persisted"

Comment (bmunkholm) [reply docs/realm.js]: @kneth Please just fix clear spelling mistakes. That's faster than mentioning them :-)

**Comment by bmunkholm** [comment]
@astigsen I assume this should be moved to "Review"?
We also need a PR or issue to make the public docs. @kneth perhaps you can add that when it fits.

**Comment by kneth** [comment]
@bmunkholm I can't find such in issue.

**Comment by kneth** [comment]
@astigsen I have readded the PR template. We need to add TypeScript and Chrome Debugging support before it can be merged.

**Comment by bmunkholm** [comment]
@kneth You ticked off "Public doc" in the checklist. Did you create an issue? Can you add link to that?

**Comment by kneth** [comment]
@bmunkholm I was too fast. I have created https://github.com/realm/realm-js/issues/1308

**Pull request merged**